### PR TITLE
Fix OOB scores in preview Random Forest; preview algorithms fixes

### PR DESCRIPTION
**Pull request by Alexsandruss**
Changes proposed in this pull request:
- Fix OOB scores in preview Random Forest: assign correct oob_error flags for oneDAL call
- Fix input data checks in preview algorithms
- Temporary fix for conversion of specific np.ndarray with OWNDATA=False flag to oneDAL table

### CODE REVIEW

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/edbf7884-28a4-f155-a04c-a4bf010d0e2e

File: sklearnex/preview/ensemble/forest.py [hunk 1]
```diff
@@ -563,14 +566,20 @@ def _onedal_gpu_supported(self, method_name, *data):
...
                 return False
+            elif sp.issparse(sample_weight):
+                return False
             elif not sample_weight:  # `sample_weight` is not supported.
                 return False
             elif not self.ccp_alpha == 0.0:
                 return False
             elif self.warm_start:
                 return False
+            elif self.oob_score:
```
Comment (samir-nasibli) [inline sklearnex/preview/ensemble/forest.py:581]: We don't support it for GPU?

Comment (Alexsandruss) [reply sklearnex/preview/ensemble/forest.py:581]: No, OOB score needed by sklearn are not computed on GPU.

File: onedal/ensemble/forest.py [hunk 2]
```diff
@@ -302,7 +302,7 @@ def _validate_targets(self, y, dtype):
         return _column_or_1d(y, warn=True).astype(dtype, copy=False)
 
     def _get_sample_weight(self, X, y, sample_weight):
-        n_samples, _ = X.shape
+        n_samples = X.shape[0]
```
Comment (KulikovNikita) [inline onedal/ensemble/forest.py:305]: Current variant directly checks for the number of dimensions

File: sklearnex/preview/decomposition/pca.py [hunk 3]
```diff
@@ -28,6 +28,8 @@
 from sklearn.utils.validation import check_array
 from sklearn.base import BaseEstimator
 from sklearn.utils.validation import check_is_fitted
+if sklearn_check_version('1.1') and not sklearn_check_version('1.2'):
+    from sklearn.utils import check_scalar
```
Comment (KulikovNikita) [inline sklearnex/preview/decomposition/pca.py:32]: Why
 these changes are in Random Forest related PR?

File: onedal/ensemble/forest.py (hunk 2 above)
Comment (Alexsandruss) [reply onedal/ensemble/forest.py:305]: Previous checked it in another way while unpacking shape

File: sklearnex/preview/decomposition/pca.py (hunk 3 above)
Comment (Alexsandruss) [reply sklearnex/preview/decomposition/pca.py:32]: Too small for separate PR

**Review by samir-nasibli — APPROVED** [review]

**Pull request merged**